src/java/main/org/apache/zookeeper/KeeperException.java

```diff
@@ -387,6 +389,8 @@ public void setCode(int code) {
         EPHEMERALONLOCALSESSION (EphemeralOnLocalSession),
         /** Attempts to remove a non-existing watcher */
         NOWATCHER (-121),
+        /** Not received packet timely */
+        TIMEOUT (-122),
```

Comment:
Similar error code needs to be added to C client to make both client library compatible.

Reply:
I think adding only error code will not be useful. Do you mean, add response timeout functionality in  C code?

Reply:
I was thinking the same error code should be added to C client so the C client can react to the new error code properly, instead of crash (the worst case). Though, I suspect properly the C client is fine as this error code will not be fired on server side (it's a pure Java client thing), so maybe we are fine without touch C client.

Reply:
REQUESTTIMEOUT sounds good to me.